pkg/search/postgres/common_test.go

```diff
@@ -262,6 +262,199 @@ func TestMultiTableQueries(t *testing.T) {
+				AddExactMatches(search.DeploymentID, uuid.NewDummy().String()).ProtoQuery(),
+			schema: deploymentBaseSchema,
+			expectedStatement: "select count(distinct(deployments.Id)) from deployments " +
+				"inner join deployments_containers on deployments.Id = deployments_containers.deployments_Id " +
+				"where (deployments.Id = $1 and deployments_containers.Image_Name_FullName = $2)",
+			expectedData: []interface{}{uuid.NewDummy(), "stackrox"},
+		},
+		{
+			desc: "base schema and child schema conjunction query on base invalid ID",
+			ctx:  baseCtx,
+			q: search.NewQueryBuilder().
+				AddExactMatches(search.ImageName, "stackrox").
+				AddExactMatches(search.DeploymentID, "not a uuid").ProtoQuery(),
+			schema: deploymentBaseSchema,
+			expectedError: `uuid: incorrect UUID length 10 in string "not a uuid"
+				value "not a uuid" in search query must be valid UUID`,
+		},
+		{
+			desc: "search of child schema mutliple results for base ID",
+			ctx:  baseCtx,
+			q: search.NewQueryBuilder().AddLinkedFieldsHighlighted(
+				[]search.FieldLabel{search.ImageName, search.EnvironmentKey},
+				[]string{search.WildcardString, search.WildcardString}).
+				ProtoQuery(),
+			schema: deploymentBaseSchema,
+			expectedStatement: "select count(distinct(deployments.Id)) from deployments " +
+				"inner join deployments_containers on deployments.Id = deployments_containers.deployments_Id " +
+				"inner join deployments_containers_envs on deployments_containers.deployments_Id = deployments_containers_envs.deployments_Id " +
+				"and deployments_containers.idx = deployments_containers_envs.deployments_containers_idx " +
+				"where (deployments_containers.Image_Name_FullName is not null and deployments_containers_envs.Key is not null)",
+		},
+		{
+			desc: "search active and inactive images with observed CVEs in non-platform deployments",
+			ctx:  baseCtx,
+			q: search.NewQueryBuilder().
+				AddExactMatches(search.VulnerabilityState, storage.VulnerabilityState_OBSERVED.String()).
+				AddStrings(search.PlatformComponent, "false", "-").
+				ProtoQuery(),
+			schema: imagesSchema,
+			expectedStatement: "select count(distinct(images.Id)) from images " +
+				"left join deployments_containers on images.Id = deployments_containers.Image_Id " +
+				"left join deployments on deployments_containers.deployments_Id = deployments.Id " +
+				"inner join image_component_edges on images.Id = image_component_edges.ImageId " +
+				"inner join image_component_cve_edges on image_component_edges.ImageComponentId = image_component_cve_edges.ImageComponentId " +
+				"inner join image_cves on image_component_cve_edges.ImageCveId = image_cves.Id " +
+				"inner join image_cve_edges on(images.Id = image_cve_edges.ImageId and image_component_cve_edges.ImageCveId = image_cve_edges.ImageCveId) " +
+				"where ((deployments.PlatformComponent = $1 or deployments.PlatformComponent is null) and (image_cve_edges.State = $2))",
```

Comment:
```suggestion
			expectedStatement: `select count(distinct(images.Id)) from images
left join deployments_containers on images.Id = deployments_containers.Image_Id
left join deployments on deployments_containers.deployments_Id = deployments.Id
inner join image_component_edges on images.Id = image_component_edges.ImageId
inner join image_component_cve_edges on image_component_edges.ImageComponentId = image_component_cve_edges.ImageComponentId
inner join image_cves on image_component_cve_edges.ImageCveId = image_cves.Id
inner join image_cve_edges on(images.Id = image_cve_edges.ImageId and image_component_cve_edges.ImageCveId = image_cve_edges.ImageCveId)
where ((deployments.PlatformComponent = $1 or deployments.PlatformComponent is null) and (image_cve_edges.State = $2))`,
```

but then we will need 
```
func normalize(s string) string {
	return strings.Join(strings.Fields(s), " ")
}
``` 
in in assertion code. I think it will be easier to read and fix without all `" +`

Reply:
Will be adressed as follow-up in #13881